api/src/main/java/edu/cornell/mannlib/vitro/webapp/edit/n3editing/VTwo/ProcessRdfForm.java

```diff
@@ -279,28 +283,40 @@ public static AdditionsAndRetractions addDependentDeletes( AdditionsAndRetractio
         return changes;
     }
 
-    public static void applyChangesToWriteModel(
-            AdditionsAndRetractions changes,
-            Model queryModel, Model writeModel, String editorUri) {
-        //side effect: modify the write model with the changes
-        Lock lock = null;
-        try{
-            lock =  writeModel.getLock();
-            lock.enterCriticalSection(Lock.WRITE);
-            if( writeModel instanceof OntModel){
-                ((OntModel)writeModel).getBaseModel().notifyEvent(new EditEvent(editorUri,true));
-            }
-            writeModel.add( changes.getAdditions() );
-            writeModel.remove( changes.getRetractions() );
-        }catch(Throwable t){
-            log.error("error adding edit change n3required model to in memory model \n"+ t.getMessage() );
-        }finally{
-            if( writeModel instanceof OntModel){
-                ((OntModel)writeModel).getBaseModel().notifyEvent(new EditEvent(editorUri,false));
-            }
-            lock.leaveCriticalSection();
-        }
+    public static void applyChangesToWriteModel(AdditionsAndRetractions changes, RDFService rdfService,
+    		String defaultGraphUri, String editorUri) {
+		appyChangesToWriteModel(changes, editorUri, rdfService, defaultGraphUri);
     }
+    
+	@Deprecated
+	public static void applyChangesToWriteModel(AdditionsAndRetractions changes, Model writeModel, String editorUri) {
+		RDFService rdfService = new RDFServiceModel(writeModel);
+		appyChangesToWriteModel(changes, editorUri, rdfService, null);
+	}
+
+	private static void appyChangesToWriteModel(AdditionsAndRetractions changes, String editorUri,
+			RDFService rdfService, String graphUri) {
+		ChangeSet cs = rdfService.manufactureChangeSet();
+		cs.addPreChangeEvent(new BulkUpdateEvent(null, true));
+		cs.addPostChangeEvent(new BulkUpdateEvent(null, false));
+		
+		ByteArrayOutputStream additionsStream = new ByteArrayOutputStream();
```

Comment:
These input and output streams should be properly closed. You could use [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html).